tests/unit/test.utils.js

```diff
@@ -606,4 +606,105 @@ describe('isCompatible', () => {
       )
     ).toBe(true);
   });
+
+  it('should be compatible if version is behind flag and no other version is available', () => {
```

Comment:
Besides these unit tests that uses fake browser-compat-data entries (which are definitely needed), in my opinion it would be good to also add at least one test case that is using a real entry from the browser-compat-data repository (just as a smoke test to increase the chances that we may notice with a test failure if the format of the compat data is being changed in a way that would break the expected behavior of the addons-linter).